### ISSUE TITLE
Display remaining battery

### ISSUE THREAD

**Issue by Benhalor** [makers-for-life/makair-control-ui#22]
Display a bar-graph on the UI showing and indication of the remaining battery. This information is only qualitative : 5 step is OK

**Comment by valeriansaliou** [comment]
Problem on that one: the firmware only provides the telemetry channel w/ a Voltage indication, which falls down very quickly as the battery approaches 0%. Hence we cannot deduce an accurate battery level % from the control UI.

**Comment by Benhalor** [comment]
An accurate level % of battery is not necessary. Only 5 states of battery is enough. It should be OK with the current voltage indication.

**Comment by valeriansaliou** [comment]
That's why we currently show voltage only, as unfortunately we do not have any BMS in the system right now, and thus cannot retrieve an accurate % value, even on 5 points.

**Comment by baptistejamin** [comment]
Legally we must provide 30 minutes of battery. 

What could be possible is to offer a count since battery mode started.

**Comment by valeriansaliou** [comment]
Given our current architecture, that would need to be handled by the firmware, not the control UI (which is enslaved to the firmware). Meaning it reflects the firmware state at the current point in time, or sends control commands as to alter its state.

**Comment by pi-r-p** [comment]
Hello,
TLDR: this will be solved with hardware v4

Battery management with hardware 3 has two problems:
- battery voltage precision is not OK. By design, we can have 800mV of error, and ADC reference can also lead to 200mV error.
- There is no "mains connected" information.

Both issues will be solved with the next hardware, so, yes, it will be possible to display 5 levels of battery. There won't be any time estimation, as it depends on battery wearing. But at least we will be able to say if we are halfway or close to no battery with a better precision.

**Comment by valeriansaliou** [comment]
![920](https://user-images.githubusercontent.com/1451907/103005213-2b2d4300-4532-11eb-9b55-69e4c9869ed9.png)
